src/components/AnalysisContent/index.js

```diff
@@ -98,26 +96,16 @@ function AnalysisContent({
         return indicator;
       });
     }
-
-    const hurumapCharts = Array.from(
-      document.querySelectorAll('div[data-chart-id]')
+    setHurumapCharts(
+      Array.from(document.querySelectorAll('div[data-chart-id]')).map(
+        element => ({
+          element,
+          geoId: element.attributes['data-geo-type'].value,
+          chartId: element.attributes['data-chart-id'].value
+        })
+      )
     );
-    if (hurumapCharts && hurumapCharts.length > 0) {
-      hurumapCharts.map(chart => {
-        const chartId = chart.attributes['data-chart-id'].value;
-        const geoId = chart.attributes['data-geo-type'].value;
-        ReactDOM.render(
-          <ApolloContainer>
-            <ThemeContainer>
-              <HURUmapChart geoId={geoId} chartId={chartId} charts={charts} />
-            </ThemeContainer>
-          </ApolloContainer>,
-          chart
-        );
-        return chart;
-      });
-    }
-  }, [charts, takwimu.country.name, topicIndex]);
+  }, [charts, takwimu.country.name]);
```

Comment:
@karimkawambwa we need `topicIndex` on this useEffect. Hurumap charts are only rendered on the initial visit & initial topic. And not rendered once you navigate from one topic to the next.